### PR TITLE
Feature/kubernetes quota

### PR DESCRIPTION
**Pull request by marie-j**
### Description of the Change

* Check quota before adding nodes
* Display prices for eligible flavors 
* Hide Windows flavor 
* Fix mobile sidebar

Several tickets related as they mostly impact the same file : 
MANAGER-1474
MANAGER-1471
MANAGER-1472
MANAGER-1501

### CODE REVIEW

File: client/app/kubernetes/translations/Messages_fr_FR.xml [hunk 1]
```diff
@@ -44,6 +44,9 @@
    <translation id="kube_nodes_add_flavor_family_vps" qtlid="505112">SANDBOX</translation>
    <translation id="kube_nodes_add_flavor" qtlid="505125">Choisissez un modèle d'instance dans le catalogue OVH public cloud</translation>
    <translation id="kube_nodes_add_pick_flavor" qtlid="505138">Choisir un modèle d'instance</translation>
+   <translation id="kube_nodes_add_flavor_QUOTA_VCPUS" qtlid="251551">Le quota de votre projet ne vous permet pas d'ajouter une machine avec autant de CPU.</translation>
```
Comment (marie-j) [inline client/app/kubernetes/translations/Messages_fr_FR.xml]: I've kept the `qtlid` as this sentence is a copy/paste from another section

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.html [hunk 2]
```diff
@@ -30,17 +30,38 @@
...
-                        data-placeholder="{{:: 'kube_nodes_add_pick_flavor' | translate }}"
-                        data-items="$ctrl.flavors"
-                        data-match="displayedName"
-                        data-disabled="!$ctrl.flavorFamily"
-                        required>
-                <span data-ng-bind="$item.displayedName"></span>
-            </oui-select>
-        </oui-field>
+        <div class="oui-field"
+             data-ng-class="{ 'oui-field_error':  (kubeNodesAdd.$submitted && !$ctrl.selectedFlavor) || $ctrl.selectedFlavor.quotaOverflow }">
```
Comment (antleblanc) [inline client/app/kubernetes/nodes/add/kubernetes-nodes-add.html]: Extra space after `:`.

File: client/app/kubernetes/nodes/kubernetes-nodes.html [hunk 3]
```diff
@@ -3,6 +3,15 @@
 </div>
 <div data-ng-if="!$ctrl.loading">
     <cui-message-container data-messages="$ctrl.messages"></cui-message-container>
+    <div class="oui-box">
+        <p class="bold" data-translate="kube_nodes_quota_description"></p>
```
Comment (antleblanc) [inline client/app/kubernetes/nodes/kubernetes-nodes.html]: Prefer using `font-weight-bold` helper. (`bold` class come from a deprecated dependency).

File: client/app/kubernetes/nodes/kubernetes-nodes.html [hunk 4]
```diff
@@ -3,6 +3,15 @@
...
+        <p write-right-required data-translate="kube_nodes_quota_more"></p>
```
Comment (antleblanc) [inline client/app/kubernetes/nodes/kubernetes-nodes.html]: +`data-`?

File: client/app/kubernetes/nodes/kubernetes-nodes.html [hunk 5]
```diff
@@ -3,6 +3,15 @@
...
+        <oui-button data-variant="primary"
+                    data-on-click="$ctrl.openSupportPopup()"
+                    write-right-required>
```
Comment (antleblanc) [inline client/app/kubernetes/nodes/kubernetes-nodes.html]: +`data-`?

File: client/app/kubernetes/translations/Messages_fr_FR.xml [hunk 6]
```diff
@@ -61,9 +64,13 @@
    <translation id="kube_nodes_flavor_error" qtlid="505281">Une erreur est survenue lors du chargement du type d'instance associé</translation>
    <translation id="kube_nodes_project_error" qtlid="505294">Une erreur s'est produite lors du chargement du projet associé</translation>
    <translation id="kube_nodes_description_project_1" qtlid="505307">Votre cluster Kubernetes utilise des nœuds pour héberger vos conteneurs et services. Tous vos nœuds seront lancés dans le projet public cloud <a class="oui-link" href="{{ url }}">{{ description }}</a> (nous permettrons de changer cela dans le futur).</translation>
-   <translation id="kube_nodes_description_project_2" qtlid="505320">Ces instances vous seront facturées normalement dans votre projet public cloud.</translation>
+   <translation id="kube_nodes_description_project_2">Ces instances sont lancées en facturation standard à l'heure. Vous pouvez retrouver les détails sur la facturation et les basculer forfait mensuel dans votre projet Public Cloud.</translation>
```
Comment (antleblanc) [inline client/app/kubernetes/translations/Messages_fr_FR.xml]: `les basculer forfait mensuel` > `les basculer EN forfait mensuel`

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js [hunk 7]
```diff
@@ -61,7 +96,16 @@ angular.module('managerApp').controller('KubernetesNodesAddCtrl', class Kubernet
       .finally(() => { this.loading = false; });
   }
 
+  instanceIsValid() {
+    return _.isNull(this.selectedFlavor.quotaOverflow);
+  }
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js]: `this.selectedFlavor.quotaOverflow === null` ?

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.html [hunk 8]
```diff
@@ -30,17 +31,40 @@
...
+                <oui-select name="flavor"
+                            id="flavor"
+                            data-model="$ctrl.selectedFlavor"
+                            data-title="{{:: 'kube_nodes_add_flavor_family' | translate }}"
+                            data-placeholder="{{:: 'kube_nodes_add_pick_flavor' | translate }}"
+                            data-items="$ctrl.flavors"
+                            data-match="displayedName"
+                            data-disabled="!$ctrl.flavorFamily"
+                            required>
+                    <button data-ng-if="$item.quotaOverflow"
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/add/kubernetes-nodes-add.html]: `type`

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.html [hunk 9]
```diff
@@ -30,17 +31,40 @@
...
-                        data-items="$ctrl.flavors"
-                        data-match="displayedName"
-                        data-disabled="!$ctrl.flavorFamily"
-                        required>
-                <span data-ng-bind="$item.displayedName"></span>
-            </oui-select>
-        </oui-field>
+        <div class="oui-field"
+             data-ng-class="{ 'oui-field_error': (kubeNodesAdd.$submitted && !$ctrl.selectedFlavor) || $ctrl.selectedFlavor.quotaOverflow }">
+            <label for="flavor" class="oui-label" data-translate="kube_nodes_add_flavor"></label>
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/add/kubernetes-nodes-add.html]: Same

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.html [hunk 10]
```diff
@@ -30,17 +31,40 @@
...
+                            data-placeholder="{{:: 'kube_nodes_add_pick_flavor' | translate }}"
+                            data-items="$ctrl.flavors"
+                            data-match="displayedName"
+                            data-disabled="!$ctrl.flavorFamily"
+                            required>
+                    <button data-ng-if="$item.quotaOverflow"
+                            class="d-flex align-items-center ui-select-choices-row oui-dropdown-option"
+                            disabled>
+                        <span data-ng-bind="$item.displayedName"></span>
+                        <span data-ng-if="$item.quotaOverflow" class="oui-icon oui-icon-info_circle d-inline position-static" aria-hidden="true"></span>
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/add/kubernetes-nodes-add.html]: Is this only for cosmetic purpose ?

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.html [hunk 11]
```diff
@@ -30,17 +31,40 @@
...
+                    </button>
+                    <div data-ng-if="!$item.quotaOverflow">
+                        <span data-ng-bind="$item.displayedName"></span>
+                        <span data-translate="kube_flavor_price" data-translate-values="{ price : $item.price }"></span>
+                    </div>
+                </oui-select>
+                <div class="oui-field__error" data-ng-if="$ctrl.selectedFlavor.quotaOverflow">
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/add/kubernetes-nodes-add.html]: `role="alert"`

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.html [hunk 12]
```diff
@@ -30,17 +31,40 @@
...
+                    <span class="oui-icon oui-icon-error" aria-hidden="true"></span>
+                    <span data-translate="{{ 'kube_nodes_add_flavor_' + $ctrl.selectedFlavor.quotaOverflow }}"></span>
+                    <button class="oui-link" data-ng-click="$ctrl.goToProjectQuota()" data-translate="kube_nodes_add_flavor_quota_manage"></button>
+                </div>
+                <div class="oui-field__error" data-ng-if="!$ctrl.selectedFlavor">
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/add/kubernetes-nodes-add.html]: `role="alert"`

File: client/app/kubernetes/translations/Messages_fr_FR.xml [hunk 13]
```diff
@@ -61,9 +65,13 @@
...
    <translation id="kube_nodes_description_project_1" qtlid="505307">Votre cluster Kubernetes utilise des nœuds pour héberger vos conteneurs et services. Tous vos nœuds seront lancés dans le projet public cloud <a class="oui-link" href="{{ url }}">{{ description }}</a> (nous permettrons de changer cela dans le futur).</translation>
-   <translation id="kube_nodes_description_project_2" qtlid="505320">Ces instances vous seront facturées normalement dans votre projet public cloud.</translation>
+   <translation id="kube_nodes_description_project_2">Ces instances sont lancées en facturation standard à l'heure. Vous pouvez retrouver les détails sur la facturation et les basculer en forfait mensuel dans votre projet Public Cloud.</translation>
+   <translation id="kube_nodes_quota_description" qtlid="251265">Pour maîtriser votre consommation Cloud, votre projet est limité en ressources.</translation>
+   <translation id="kube_nodes_quota_more" qtlid="251278">Si vous avez besoin de plus de ressources, vous pouvez les demander à notre équipe en cliquant sur le bouton ci dessous.</translation>
```
Comment (FredericEspiau) [inline client/app/kubernetes/translations/Messages_fr_FR.xml]: `ci-dessous` (you don't have to remove the `qtlid`)

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.html [hunk 14]
```diff
@@ -42,18 +42,11 @@
...
-                        <span class="oui-icon oui-icon-info_circle d-inline position-static"
-                              aria-label="{{:: 'kube_nodes_add_flavor_' + $item.quotaOverflow | translate }}">
-                        </span>
-                    </button>
-                    <div data-ng-if="!$item.quotaOverflow">
-                        <span data-ng-bind="$item.displayedName"></span>
-                        <span data-translate="kube_flavor_price" data-translate-values="{ price : $item.price }"></span>
-                    </div>
+                            <div class="d-flex align-items-center">
+                                <span data-ng-if="$item.quotaOverflow" class="oui-icon oui-icon-warning_circle d-inline position-static" aria-hidden="true"></span>
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/add/kubernetes-nodes-add.html]: -`.d-inline` ?

Comment (marie-j) [reply client/app/kubernetes/nodes/add/kubernetes-nodes-add.html]: The `oui-select` uses special style for `oui-icon` that hides them by default (`display: none`)

**Pull request merged**